src/Uno.UI/UI/Xaml/Controls/Popup/PopupPanel.cs

```diff
@@ -97,7 +97,10 @@ protected override Size ArrangeOverride(Size finalSize)
 	{
 		// Note: Here finalSize is expected to be the be the size of the window
 
-		var size = _lastMeasuredSize;
+		var child = this.GetChildren().FirstOrDefault();
+
+		// the Child could have been measured without PopupPanel itself being measured, so _lastMeasuredSize could be outdated
+		var size = Popup.CustomLayouter is { } || child is null ? _lastMeasuredSize : LayoutInformation.GetDesiredSize(child);
```

Comment:
I'm worried about the performance characteristics of this change.
Are we able to detect exactly when this measurement becomes outdated and invalidate the measurement only for this case?

Reply:
This shouldn't affect performance pretty much at all. The call to GetDesiredSize gets an already measured size, and doesn't do any remeasurements. Anyway, I'm working on a different solution